assert/assert.go

```diff
@@ -197,7 +199,7 @@ func NilError(t TestingT, err error, msgAndArgs ...interface{}) {
 // called from the goroutine running the test function, not from other
 // goroutines created during the test. Use Check with cmp.Equal from other
 // goroutines.
-func Equal(t TestingT, x, y interface{}, msgAndArgs ...interface{}) {
+func Equal[ANY any](t TestingT, x, y ANY, msgAndArgs ...any) {
```

Comment:
I wonder if this one could/should be the `comparable` constraint?

Reply:
Good question! I did attempt to use `comparable` here while working on this. It works for some types, but a few tests failed. Currently `Equal` works with pointers and interfaces like `error`. Changing to `comparable` caused the build to fail because `error` is not comparable. So I had to use `any`.